File: Compose/YABA/app/src/main/java/dev/subfly/yaba/core/state/detail/linkmark/LinkmarkDetailEvent.kt
package dev.subfly.yaba.core.state.detail.linkmark

import dev.subfly.yaba.core.model.utils.ReaderFontSize
import dev.subfly.yaba.core.model.utils.ReaderLineHeight
import dev.subfly.yaba.core.model.utils.ReaderTheme
import dev.subfly.yaba.core.webview.WebShellLoadResult

sealed interface LinkmarkDetailEvent {
    data class OnInit(val bookmarkId: String) : LinkmarkDetailEvent
    data class OnReaderWebInitialContentLoad(val result: WebShellLoadResult) : LinkmarkDetailEvent
    data object OnDeleteBookmark : LinkmarkDetailEvent
    data object OnToggleReaderTheme : LinkmarkDetailEvent
    data object OnToggleReaderFontSize : LinkmarkDetailEvent
    data object OnToggleReaderLineHeight : LinkmarkDetailEvent
    data class OnSetReaderTheme(val theme: ReaderTheme) : LinkmarkDetailEvent
    data class OnSetReaderFontSize(val fontSize: ReaderFontSize) : LinkmarkDetailEvent
    data class OnSetReaderLineHeight(val lineHeight: ReaderLineHeight) : LinkmarkDetailEvent
    data object OnRequestNotificationPermission : LinkmarkDetailEvent
    data class OnScheduleReminder(
        val selectedDateMillis: Long,
        val hour: Int,
        val minute: Int,
        val title: String,
        val message: String,
    ) : LinkmarkDetailEvent
    data object OnCancelReminder : LinkmarkDetailEvent
    data class OnExportMarkdownReady(val markdown: String) : LinkmarkDetailEvent
    data class OnExportPdfReady(val pdfBase64: String) : LinkmarkDetailEvent
}

File: Darwin/YABACore/State/Detail/Linkmark/LinkmarkDetailEvent.swift
//
//  LinkmarkDetailEvent.swift
//  YABACore
//
//  Parity with Compose `LinkmarkDetailEvent` — complex payloads use `Data`/`String` for host bridges.
//  Reminder `titleKey` / `messageKey` are iOS `Localizable.strings` keys (resolved in `ReminderManager`).
//

import Foundation

public enum LinkmarkDetailEvent: Sendable {
    case onInit(bookmarkId: String)
    /// Call when the UI has resolved the link URL (needed for refresh).
    case onLinkSourceUrl(String)
    case onSaveReadableContent(html: Data)
    case onReaderWebInitialContentLoad(resultJson: String?)
    case onDeleteBookmark(bookmarkId: String)
    case onToggleReaderTheme
    case onToggleReaderFontSize
    case onToggleReaderLineHeight
    case onSetReaderTheme(ReaderTheme)
    case onSetReaderFontSize(ReaderFontSize)
    case onSetReaderLineHeight(ReaderLineHeight)
    case onRequestNotificationPermission
    case onScheduleReminder(fireAt: Date, titleKey: String, messageKey: String)
    case onCancelReminder
    case onExportMarkdownReady(String)

    case updateLinkMetadata(
        bookmarkId: String,
        url: String,
        domain: String?,
        videoUrl: String?,
        audioUrl: String?,
        metadataTitle: String?,
        metadataDescription: String?,
        metadataAuthor: String?,
        metadataDate: String?
    )
}
